shared-libs/settings/src/index.js

```diff
@@ -41,7 +41,16 @@ const getKey = () => {
   const url = `${getCouchConfigUrl()}/couch_httpd_auth/secret`;
   return request
     .get(url, { json: true })
-    .then(key => Buffer.from(key).slice(0, KEY_LENGTH));
+    .then(key => {
+      let buffer = Buffer.from(key);
+      if (buffer.length < KEY_LENGTH) {
+        const duplicateCount = Math.ceil(KEY_LENGTH / buffer.length);
```

Comment:
Is it possible for buffer.length to be 0, causing a divide by zero error?

Reply:
I pushed a commit to have the code throw an explicit error when the secret is empty.